ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -1717,7 +1717,7 @@ private long checkInconsistentAppendEntries(TermIndex previous, List<LogEntryPro
     }
 
     // Check if "previous" is contained in current state.
-    if (previous != null && !state.containsTermIndex(previous)) {
+    if (previous != null && !state.containsTermIndex(previous) && appendLogFuture.get().isDone()) {
```

Comment:
This will disable inconsistent check when the server is busy (appendLogFuture.get().isDone() will mostly false).  We need to remember the entries being appended to the log; see https://issues.apache.org/jira/secure/attachment/13074189/1215_review.patch